src/internal/observable/generate.ts

```diff
@@ -344,13 +345,12 @@ export function generate<T, S>(
   // TODO: Remove this as we move away from deprecated signatures
   // and move towards a configuration object argument.
   if (arguments.length == 1) {
-    const options = initialStateOrOptions as GenerateOptions<T, S>;
-    initialState = options.initialState;
-    condition = options.condition;
-    iterate = options.iterate;
-    resultSelector = options.resultSelector || (identity as ResultFunc<S, T>);
-    scheduler = options.scheduler;
+    // If we only have one argument, we can assume it is a configuration object.
+    // Note that folks not using TypeScript may trip over this.
+    ({ initialState, condition, iterate, resultSelector = identity as any, scheduler } = initialStateOrOptions as GenerateOptions<T, S>);
```

Comment:
nitpick: I think this assertion could be narrower than `any`. Using `identity as ResultFunc<S, T>` should work. If not, `identity as ResultFunc<any, T>`